service/history/task/cross_cluster_task.go

```diff
@@ -470,6 +486,90 @@ func (t *crossClusterSourceTask) GetCrossClusterRequest() (request *types.CrossC
 	return request, nil
 }
 
+func (t *crossClusterSourceTask) VerifyLastWriteVersion(
+	mutableState execution.MutableState,
+	taskInfo *persistence.CrossClusterTaskInfo,
+) (bool, error) {
+	lastWriteVersion, err := mutableState.GetLastWriteVersion()
+	if err != nil {
+		return false, err
+	}
+	ok, err := verifyTaskVersion(t.shard, t.logger, taskInfo.DomainID, lastWriteVersion, taskInfo.Version, taskInfo)
+	if err != nil || !ok {
+		return false, err
+	}
+	return true, nil
+}
+
+func (t *crossClusterSourceTask) getRequestForApplyParentPolicy(
+	ctx context.Context,
+	taskInfo *persistence.CrossClusterTaskInfo,
+	mutableState execution.MutableState,
+) (*types.CrossClusterApplyParentClosePolicyRequestAttributes, processingState, error) {
+
+	// No need to check the target failovers, only the active cluster should poll tasks
+	// if active cluster changes during polling, target should return error to the source
+	verified, err := t.VerifyLastWriteVersion(mutableState, taskInfo)
+	if err != nil || !verified {
+		return nil, t.processingState, err
+	}
+
+	attributes := &types.CrossClusterApplyParentClosePolicyRequestAttributes{}
+	children := mutableState.GetPendingChildExecutionInfos()
+	for _, childInfo := range children {
+		targetDomainEntry, err := t.shard.GetDomainCache().GetDomain(childInfo.DomainName)
+		if err != nil {
+			return nil, t.processingState, err
+		}
+		targetCluster := targetDomainEntry.GetReplicationConfig().ActiveClusterName
+		if targetCluster == t.targetCluster {
```

Comment:
Looks like if they are not equal, that child will be dropped? 

I am thinking the case where there're two childworkflows in two different domains, one of them did a failover while we are fetching the task and causing `targetCluster != t.targetCluster`, I guess in this case, we will need to create a new task for this failed over domain. 
We can still send the child to the target cluster (by removing the check) and when receiving the domain not active error, source task executor will create the new task. Or mark the task as invalid here and return an error, again the source task executor will split the task into two.

Another idea is that when converting close execution task to apply parent close task, we can generate one applyParentClose task for each target (child) domain, so we'll never need to split the task when processing it.